tools/coqdoc/cmdArgs.ml

```diff
@@ -78,13 +78,13 @@ let arg_path f = Arg.String (fun s ->
   else
     Arg.current := !Arg.current + 1;
     prefs := f !prefs (normalize_path s, Sys.argv.(!Arg.current + 1)))
-let arg_url_path f = Arg.String (fun s ->
+let arg_url_path f = Arg.String (fun url ->
   if Array.length Sys.argv < !Arg.current + 3 ||
     Sys.argv.(!Arg.current + 2).[0] = '-' then
     raise (Arg.Bad ("Two arguments expected: <url> and <path>"))
   else
     Arg.current := !Arg.current + 1;
-    f s Sys.argv.(!Arg.current + 1))
+    f Sys.argv.(!Arg.current + 1) url)
```

Comment:
While the fix is fine, I think the resulting code will be very confusing for anyone that reads it later on, since it surreptitiously swaps the arguments. Rather than modifying the function above, it might better to just replace its call `arg_url_path Index.add_external_library` below by `arg_url_path (fun url lp -> Index.add_external_library lp url)`.

Reply:
To have to twist less our minds, we could also swap the arguments in `Index.add_external_library`?

Reply:
You could also use labels, they're a great way to disambiguate arguments with the same type.

Reply:
> To have to twist less our minds, we could also swap the arguments in `Index.add_external_library`?

I did consider it, but since it involves changing the API, I refrained from suggesting it. (And I consider the API to be fine here, since having the logical path as the first argument is the sensible choice, in my opinion. As a user, I did mess a few times the order of the arguments of `--external` becomes it is not natural to me.) That said, I would not oppose it, since I doubt anyone other than Coq itself is using that API.

Reply:
OK, then, to preserve the API, I adopted `arg_url_path (fun url lp -> Index.add_external_library lp url)`.